### ISSUE TITLE
ClickHouse Error Logs Not Written to clickhouse-server.err.log

### ISSUE THREAD

**Issue by shiwai** [ClickHouse/ClickHouse#49167]
Hello ClickHouse team,

I am currently using ClickHouse server version 22.8.6 revision 54460 and have encountered a strange issue with the error logging behavior. The error logs are being written to clickhouse-server.log instead of clickhouse-server.err.log.

I have used the lsof command to investigate this issue and found the following results:

```shell
root@clickhouse01:~# lsof /var/log/clickhouse-server/clickhouse-server.log
COMMAND     PID       USER   FD   TYPE DEVICE  SIZE/OFF     NODE NAME
clickhous 38380 clickhouse  725w   REG 259,25 953403933 13238294 /var/log/clickhouse-server/clickhouse-server.log

root@clickhouse01:~# lsof /var/log/clickhouse-server/clickhouse-server.err.log
(no output)
```

As seen in the results above, the ClickHouse process is not using the clickhouse-server.err.log file.

I haven't made any changes to the logging configuration in my config.xml file, which is as follows:

```shell
<level>trace</level>
<log>/var/log/clickhouse-server/clickhouse-server.log</log>
<errorlog>/var/log/clickhouse-server/clickhouse-server.err.log</errorlog>
<!-- Rotation policy
     See https://github.com/pocoproject/poco/blob/poco-1.9.4-release/Foundation/include/Poco/FileChannel.h#L54-L85
  -->
<size>1000M</size>
<count>10</count>
```

Could you please help me understand what might cause this issue, and how to resolve it? Thank you in advance!

Best regards,

**Comment by CheSema** [comment]
I haven't encountered such behaviour. So I unable to provide solution. But I will try to ask some question, hope it could help in investigation.

How stable is this behavior? Did you try to restart?

Do you see that row in stderr at start?
`        std::cerr << "Logging errors to " << errorlog_path << ext << std::endl;`
https://github.com/ClickHouse/ClickHouse/blob/master/src/Loggers/Loggers.cpp#L129

Is there sufficient space on a volume with logs?

Do you define the command param `--logger.errorlog` at cmd when run server?

Are there rotated files present in the directory with older errors logs? How old are they?

Is there any other process that could remove or use that path? An other CH instance, for example.

**Comment by shiwai** [comment]
@CheSema Thank you for your response!

I'd like to provide some additional information based on your questions.

1. My ClickHouse instance is being used in a production environment. Initially, error logs were properly written to clickhouse-server.err.log. However, one day, logs suddenly stopped being written to this file, even though I did not restart ClickHouse or make any configuration changes that day.
2. Since ClickHouse is used in a production environment, I cannot restart it casually. I plan to find a maintenance window in the near future to perform a restart and see if it resolves the issue.
3. There is plenty of space available for log writing, with 200GB of remaining space on the volume.
4. I started ClickHouse using systemctl start clickhouse-server.service without specifying --logger.errorlog.
5. Currently, there is a clickhouse-server.err.log file with a size of 594MB.
6. Only one ClickHouse instance is running on this machine.


This issue isn't a significant problem, so there is no need to spend too much effort investigating it.

I'd like to ask if there's a way to reactivate ClickHouse's use of this file without restarting the instance. For example, in MySQL, I can use set global slow_query_log_file = '/path/mysqld_slow.log' to change the directory where logs are printed.

Based on the current situation, it seems that finding a maintenance window to perform a restart should help restore the logging functionality. Thank you.
